### ISSUE TITLE
✨ Disable looping in certain apps

### ISSUE THREAD

**Issue by frieza213** [MrKai77/Loop#233]
### Feature Request Description

So i don't really want arc to be looping and i think there must be some apps that everyone have which they don't wanna snap, so it would be very nice to have a feature to disable snapping for certain apps

### Final Checks

- [X] My issue title is descriptive
- [X] This is a single feature request (multiple feature requests should be reported individually)

**Comment by JaceThings** [comment]
> think there must be some apps that everyone have which they don't wanna snap

🫣 Wouldn't you just... not activate the loop for that app? I've never had the loop activate on something I didn't want it to.

**Comment by frieza213** [comment]
@Soft-Bred no but like with the mouse drag not keyboard shortcut

**Comment by JaceThings** [comment]
ah, makes more sense, i have that turned off, too slow 💀

**Comment by frieza213** [comment]
i use three finger drag when browsing since i am using trackpad to scroll and without using keyboard i can snap but yeah it is slow unless you are using the trackpad actively

**Comment by MrKai77** [comment]
Got it, I can look into this!
But I am prioritizing other issues at the moment so this may take a while to implement :)

**Comment by frieza213** [comment]
no problem mate

**Comment by Dhairy2003** [comment]
@MrKai77  Can u introduce shortcuts to move and resize window too ?? kinda similar to rectangle pro

**Comment by MrKai77** [comment]
> @MrKai77 Can u introduce shortcuts to move and resize window too ?? kinda similar to rectangle pro

@Dhairy2003 Please open a new issue for this, as it sounds unrelated to this current one :)

**Comment by just-doit** [comment]
I use a virtual machine (parallels, windows guest) in which some of the key combinations for looping have a different meaning. It would also be easiest here if I could exclude the app, as it always runs in full screen.

**Comment by just-doit** [comment]
I have already experimented a little myself ;-)
The idea is that there is a UserDefaults entry (string array) which contains the bundle IDs of the apps for which the trigger key should be ignored.
It could look like this in the settings:
<img width="612" alt="Bildschirmfoto 2024-03-12 um 14 39 49" src="https://github.com/MrKai77/Loop/assets/4004387/5a25bbab-edb2-45bd-b57e-3075e8abbbdf">

The `+` button shows a menu with a list of apps installed in the Applications folder.
<img width="397" alt="Bildschirmfoto 2024-03-12 um 14 40 22" src="https://github.com/MrKai77/Loop/assets/4004387/49ac0886-b032-4c0e-9625-53038c03fa9c">

In order to ignore those apps I modified the `WindowEngine.frontmostWindow` function:
```swift
    static var frontmostWindow: Window? {
        guard
            let app = NSWorkspace.shared.runningApplications.first(where: { $0.isActive }),
            !Defaults[.applicationBlackList].contains(app.bundleIdentifier ?? ""),
            let window = Window(pid: app.processIdentifier)
        else {
            return nil
        }
        return window
    }
```

If you're interested I could open a pull request. ;-)
